inst/include/Rcpp/exceptions.h

```diff
@@ -126,18 +126,47 @@ namespace Rcpp{
 
 } // namespace Rcpp
 
+inline SEXP nth(SEXP s, int n) {
+  return Rf_length(s) > n ? (n == 0 ? CAR(s) : CAR(Rf_nthcdr(s, n))) : R_NilValue;
+}
+
 inline SEXP get_last_call(){
-    SEXP sys_calls_symbol = Rf_install( "sys.calls" ) ;
-    Rcpp::Shield<SEXP> sys_calls_expr( Rf_lang1(sys_calls_symbol) );
-    Rcpp::Shield<SEXP> calls( Rcpp_eval( sys_calls_expr, R_GlobalEnv ) );
-    SEXP res = calls ;
-    while( !Rf_isNull(CDR(res)) ) res = CDR(res);
-    return CAR(res) ;
+    SEXP sys_calls_symbol = Rf_install("sys.calls");
+    SEXP identity_symbol = Rf_install("identity");
+    SEXP identity_fun = Rf_findFun(identity_symbol, R_BaseEnv);
+    SEXP tryCatch_symbol = Rf_install("tryCatch");
+    SEXP evalq_symbol = Rf_install("evalq");
+
+    Rcpp::Shield<SEXP> sys_calls_expr(Rf_lang1(sys_calls_symbol));
+    Rcpp::Shield<SEXP> calls(Rcpp_eval(sys_calls_expr, R_GlobalEnv));
+
+    SEXP cur, prev;
+    prev = cur = calls;
+    while(CDR(cur) != R_NilValue) {
+      SEXP expr = CAR(cur);
+
+      // We want the call just prior to the call from Rcpp_eval
+      // This conditional matches
+      // tryCatch(evalq(sys.calls(), .GlobalEnv), error = identity, interrupt = identity)
+      if (TYPEOF(expr) == LANGSXP &&
```

Comment:
Done in https://github.com/RcppCore/Rcpp/pull/582/commits/9412f1f3d2ca36637709fa59d9fc816b2421de73